terraform/main.tf

```diff
@@ -2,15 +2,40 @@
 # See LICENSE file for licensing details.
 
 resource "juju_application" "mongodb" {
-  name  = var.app_name
-  model = var.model_name
 
   charm {
-    name    = "mongodb"
-    channel = var.channel
-    base    = "ubuntu@22.04"
+    name     = "mongodb"
+    channel  = var.channel
+    revision = var.revision
+    base     = var.base
   }
-  config = var.config
-  units  = 1
-  trust  = true
-}
+  config      = var.config
+  model       = var.model
+  name        = var.app_name
+  units       = var.units
+  constraints = var.constraints
+
+
+  # TODO: uncomment once final fixes have been added for:
+  # Error: juju/terraform-provider-juju#443, juju/terraform-provider-juju#182
+  # placement = join(",", var.machines)
+
+  endpoint_bindings = [
+    for k, v in var.endpoint_bindings : {
+      endpoint = k, space = v
+    }
+  ]
+
+  storage_directives = var.storage
+
+  lifecycle {
+    precondition {
+      condition     = length(var.machines) == 0 || length(var.machines) == var.units
+      error_message = "Machine count does not match unit count"
+    }
+    precondition {
+      condition     = length(var.storage) == 0 || lookup(var.storage, "count", 0) <= 1
+      error_message = "Only one storage is supported"
```

Comment:
**question:** Is this condition true for Mongo? Do you only support 1 storage or can you have multiple?

Reply:
In MongoDB-VM we only support one storage, but in K8s we support two storages.

The reason we don't yet support that in MongoDB VM is because we aren't sure how to support that in the case of an upgrade

Reply:
Is that also true? We can't have more than one storage PVC on K8s. Or do you mean multiple storage PVCs, but only 1 of each 'named' one?